### PR TITLE
Fix backslash escaping inside double quotes

### LINKED ISSUE #536: How to escape spaces in browser path
**Newsboat version**:

    newsboat 2.15.0 - https://newsboat.org/
    Copyright (C) 2006-2015 Andreas Krennmair
    Copyright (C) 2015-2019 Alexander Batischev
    Copyright (C) 2006-2017 Newsbeuter contributors
    Copyright (C) 2017-2019 Newsboat contributors

    Newsboat is free software licensed under the MIT License. (Type `newsboat -vv' to see the full text.)
    It bundles JSON for Modern C++ library, licensed under the MIT License: https://github.com/nlohmann/json
    It bundles an alphanum algorithm implementation licensed under the MIT license: http://www.davekoelle.com/alphanum.html

    newsboat 2.15.0
    System: Darwin 18.2.0 (x86_64)
    Compiler: g++ 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.4)
    ncurses: ncurses 5.7.20081102 (compiled with 5.7)
    libcurl: libcurl/7.54.0 LibreSSL/2.6.4 zlib/1.2.11 nghttp2/1.24.1 (compiled with 7.54.0)
    SQLite: 3.24.0 (compiled with 3.24.0)
    libxml2: compiled with 2.9.4

I want to set the browser command to `/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --app %u`. The `--app` parameter opens Chrome in app mode on that page, meaning no bookmarks, tabs or address bar. Here are the config lines that I have tried and the errors they give:

* `browser /Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --app %u`: 

    `sh: /Applications/Google https://news.ycombinator.com/item?id=20077421: No such file or directory`

* `browser '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome' --app %u`: 
    
    ` sh: -c: line 0: unexpected EOF while looking for matching `''`
     `sh: -c: line 1: syntax error: unexpected end of file`

* `browser /Applications/Google\\ Chrome.app/Contents/MacOS/Google\\ Chrome --app %u`:
 
    `sh: /Applications/Google\: No such file or directory`

The OP in #172 relied instead on using `/usr/bin/open`, but I am unable to get that to work correctly with the `--app` argument.

How do I escape the spaces in the browser executable path?

### LINKED ISSUE #642: How to use regular expression to highlight/ignore
<details><summary>Newsboat version</summary>

```
System: Darwin 18.7.0 (x86_64)
Compiler: g++ 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.4)
ncurses: ncurses 5.7.20081102 (compiled with 5.7)
libcurl: libcurl/7.54.0 LibreSSL/2.6.5 zlib/1.2.11 nghttp2/1.24.1 (compiled with 7.54.0)
SQLite: 3.24.0 (compiled with 3.24.0)
libxml2: compiled with 2.9.4
```
</details>

I've been trying to configure ignoring/highlighting articles based on regular expressions without success.

For example `/\bgit\b/i` should match any appearance of the word "git", but not when it's part of a bigger word like `digital`. 

But neither of this work and I can't figure out why :/
```
ignore-article "*" "title =~ \"/\bfacebook\b/i\""
highlight-article "title =~ \"/\bgit\b/i\"" white blue bold
```

Could you either help me with the regular expression or point me to any resource that I can follow to get it right? Thanks!

### LINKED ISSUE #926: Can't use `sed` in urls file
Here is the content of an `urls` file:

```
"exec:curl --silent https://feeds.metaebene.me/raumzeit/m4a  | sed 's#\(</guid>\|</id>\)#-M4A&#'"    "~Raumzeit [M4A]"
"exec:curl --silent https://feeds.metaebene.me/raumzeit/mp3  | sed 's#\(</guid>\|</id>\)#-MP3&#'"    "~Raumzeit [MP3]"
"exec:curl --silent https://feeds.metaebene.me/raumzeit/oga  | sed 's#\(</guid>\|</id>\)#-Vorbis&#'" "~Raumzeit [Vorbis]"
"exec:curl --silent https://feeds.metaebene.me/raumzeit/opus | sed 's#\(</guid>\|</id>\)#-Opus&#'"   "~Raumzeit [Opus]"
```

As far as I know, as an uniq identifier, RSS 2.0 uses `<guid>` and Atom `<id>`. The sed command includes both. It simply inserts a string between the element's content and the respective end-tag.

_Originally posted by @mglh in https://github.com/newsboat/newsboat/issues/898#issuecomment-626187889_

@Minoru reproduced on current HEAD:

newsboat r2.19-262-gb188 - https://newsboat.org/
Copyright (C) 2006-2015 Andreas Krennmair
Copyright (C) 2015-2020 Alexander Batischev
Copyright (C) 2006-2017 Newsbeuter contributors
Copyright (C) 2017-2020 Newsboat contributors

Newsboat is free software licensed under the MIT License. (Type `./newsboat -vv' to see the full text.)
It bundles:
- JSON for Modern C++ library, licensed under the MIT License: https://github.com/nlohmann/json
- optional-lite library, licensed under the Boost Software License: https://github.com/martinmoene/optional-lite

newsboat r2.19-262-gb188
System: Linux 5.6.0-1-amd64 (x86_64)
Compiler: g++ 9.3.0
ncurses: ncurses 6.2.20200212 (compiled with 6.2)
libcurl: libcurl/7.68.0 GnuTLS/3.6.13 zlib/1.2.11 brotli/1.0.7 libidn2/2.3.0 libpsl/0.21.0 (+libidn2/2.3.0) libssh2/1.8.0 nghttp2/1.40.0 librtmp/2.3 (compiled with 7.68.0)
SQLite: 3.31.1 (compiled with 3.31.1)
libxml2: compiled with 2.9.10

### PR DESCRIPTION
**Pull request by Minoru**
This fixes the issue where every other time, backslash had to be escaped three times instead of one, i.e. to get `\ \ \ \` (four backslashes separated by spaces) one had to write `\\ \\\\ \\ \\\\`. For a detailed explanation of the problem and the solution, see [the commit message here](https://github.com/newsboat/newsboat/commit/8f6b73a33036bb05084c36a9f445c00ed32c672a).

This PR also:
- documents double quotes and how escapes in them work (it doesn't mention `\n`, `\t`, `\v`, and `\r` because #928) (@der-lyse, that's the part I'd like you to take a look at);
- performs a little refactoring of the offending function;
- of course, adds tests for all the issue that this fixes.

Fixes #536
Fixes #642 
Fixes #926

Reviews from everyone are welcome. I'll merge this in three days.

### CODE REVIEW

File: doc/newsboat.asciidoc [hunk 1]
```diff
@@ -890,8 +919,8 @@ This feature is often used to create a feed that contains all unread articles:
 
 	"query:Unread Articles:unread = \"yes\""
 
-Note the quotes that are necessary around the complete query "URL" and the
-backslashes that are necessary to escape the quotes in the filter expression.
+Note the <<_using_double_quotes,use of double quotes>> to preserve spaces in
```
Comment (der-lyse) [inline doc/newsboat.asciidoc:922]: I'm a little bit in between. I really liked the two explicit hints on double quotes and their escapes, but this change makes it consistent with other paragraphs and even links to a much more detailed explanation. Let's keep it this way.

**Comment by Minoru** [comment]
(I haven't abandoned this, just keep getting distracted by incoming issues.)

File: src/utils.cpp [hunk 2]
```diff
@@ -136,40 +136,30 @@ std::vector<std::string> utils::tokenize_quoted(const std::string& str,
...
-						if (str[last_pos - 1] == '\\') {
-							if (attach_backslash) {
-								token.append("\\");
-							}
-							attach_backslash = !attach_backslash;
-						}
+						// We checked that there are no unbalanced backslashes
+						// inside the quoted string, so it's fine to access
+						// `last_pos + 1` here
+						append_escapes(token, str[last_pos + 1]);
```
[the remark this answers is not in the document]
Comment (Minoru) [reply src/utils.cpp]: Yeah, I didn't think it through well enough. I'll add an `if` around that line. It's not a given that users will always have completely correct configs, so I think it'd pay to be a bit defensive here.

**Comment by Minoru** [comment]
Thanks for the reviews!

Rebasing to squash the fixup commit…

**Pull request merged**